src/controllers/ussd_controller.ts

```diff
@@ -0,0 +1,58 @@
+import { Request, Response } from "express";
+// import Ussd from '../model/ussd_schema';
+
+
+export const getRequest = (req: Request, res: Response) => {
+  res.send(
+    "Build a simple USSD-BASED service that allows a user to update his name via USSD."
+  );
+};
+
+export const postRequest = ( req: Request, res: Response) => {
+
+  let { sessionId, serviceCode, phoneNumber } = req.body;
+  let text: string = req.body;
+
+  console.log(text)
+  let response: string = "";
+
+  if (!text) {
+    response = `CON Enter your fullname`;
+    res.send(response);
+  }
+  if (text) {
```

Comment:
I think this conditional will evaluate to true for all webhooks coming from the AfricaTalking provider;
Making it the last (default) conditional block instead of the first should fix the problem, something like this;


  if (!text) {
    response = `CON Enter your fullname`;
    res.send(response);
  }
   if (text === "1") {
    response = `CON Choose account information you want to view
        1. Account Name
        2. Account Number
        3. Account Balance`;
    res.send(response);
  } else if (text === "2") {
    response = `END Your phone number is ${phoneNumber}`;
    res.send(response);
  } else if (text === "1*1") {
    let accountName = "Steve Chude";
    response = `END Your account name is ${accountName}`;
    res.send(response);
  } else if (text === "1*2") {
    let accountNumber = generateAccountNumber();
    response = `END Your account number is ${accountNumber}`;
    res.send(response);
  } else if (text === "1*3") {
    let balance = "NGN 10,000";
    response = `END Your account balance is ${balance}`;
    res.send(response);
  } else {
    response = `CON What do you want to check
      1. My Account
      2. My Phone Number`;
    res.send(response);
  }

Reply:
Wow, thank you so much Ayuba. I would implement this your thought process now. i'm sure this would work. I really appreciate.